src/components/draw/DrawDirective.js

```diff
@@ -315,6 +317,9 @@ goog.require('ga_styles_service');
           }
         };
 
+        // This variable keeps track of the existence or not of the layer before
+        // activation of the draw tool (kml loaded from permalink)
+        var wasNotOnMap;
```

Comment:
loaded via permalink is an example it could be another component. I think the comment is enough to understand what I meant creating this variable.